src/routes/loyalty.js

```diff
@@ -1,17 +1,32 @@
 var express = require('express');
 
-var tradeshift = require('../services/TradeshiftAPIClient');
+var TradeshiftAPIClient = require('../services/TradeshiftAPIClient');
 
 var router = express.Router();
 
-router.post('/document/:documentId/thank', function(request, response) {
-	var documentId = request.params.documentId;
+router.post('/document/:documentReferenceId/thank', function(request, response) {
+	var documentReferenceId = request.params.documentReferenceId;
 	var documentReferenceNumber = request.body.businessId;
 	var message = request.body.message;
-	// post comment
-	// add document tag
-	console.log('thanking', documentId, documentReferenceNumber, message);
-	response.status(200).end();
+
+	console.log('thanking', documentReferenceId, documentReferenceNumber, message);
+
+	var ts = new TradeshiftAPIClient(request.session.auth.accessToken);
+	ts.addCommentToDocument({
+		documentReferenceId: documentReferenceId,
+		documentReferenceNumber: documentReferenceNumber,
+		message: message,
+	}).then(function(result) {
+		return ts.addTagToDocument({
+			documentId: documentReferenceId,
+			tag: 'loyalty',
+		});
+	}).then(function(result) {
+		response.status(200).send(result);
+	}).catch(function(error) {
+		console.log('error is', error);
+		response.status(500).send(error);
+	}).done();
```

Comment:
This will post a comment on the document _and_ add a tag to the document so we can keep track of which have been thanked, and the ```GET /documents``` call used to build the initial view filters out the tagged documents. (So maybe the tag adding here (or the tag filtering elsewhere) should be commented out during local development to allow things to move more quickly.) @schuurthing